script/pip_check

```diff
@@ -3,7 +3,7 @@ PIP_CACHE=$1
 
 # Number of existing dependency conflicts
 # Update if a PR resolve one!
-DEPENDENCY_CONFLICTS=3
+DEPENDENCY_CONFLICTS=4
```

Comment:
you are not allowed to increase conflicts.

Reply:
I'll see what I can do to resolve this conflict; what is the generally-accepted path forward when a dependency has sub-dependencies that are in conflict with other home assistant components, except to update this value?

Reply:
I've created [this PR](https://github.com/poljar/matrix-nio/pull/366/files) with `matrix-nio` in an attempt to bring the dependencies in a compatible state, though it seems that the way that pip checks for compatibility is incompatible with poetry's `^` operator:
e.g. `h11` `0.14.0` is not considered in range py `pip` based on `h11 = "^0.12.0"`.